File: schemas/media-streaming/drizzle/labels.ts
// labels: Record labels that distribute and publish albums.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";

export const labels = pgTable(
  "labels",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    slug: text("slug").unique().notNull(),
    website: text("website"),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_labels_name").on(table.name),
  ]
);

File: schemas/media-streaming/spacetimedb/labels.rs
// labels: Record labels that publish and distribute music albums.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = labels, public)]
pub struct Label {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub name: String,
    #[unique]
    pub slug: String,
    pub website: Option<String>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
